pkg/framework/test/etcd.go

```diff
@@ -12,10 +13,10 @@ import (
 
 // Etcd knows how to run an etcd server. Set it up with the path to a precompiled binary.
 type Etcd struct {
-	// The path to the etcd binary
 	Path           string
 	EtcdURL        string
-	session        *gexec.Session
+	ProcessStarter simpleSessionStarter
```

Comment:
I find it odd to have the "ProcessStarter" as a member of the object, rather than the other way around (I would expect the ProcessStarter to take the object to start as an argument).

Reply:
I see your point. To be honest, we never thought about it that way. Let me think about that, and discuss that with @totherme tomorrow.